lib/template-dependencies.js

```diff
@@ -119,21 +119,29 @@ function solve(options, elmVersion, elmJson, extra, onlineFirst) {
   }
 
   try {
-    return JSON.parse(
-      onlineFirst && !options.offline
-        ? dependencyProvider.solveOnline(options, elmVersion, elmJson, extra)
-        : dependencyProvider.solveOffline(options, elmVersion, elmJson, extra)
+    const dependencies = /** @type {unknown} */ (
+      JSON.parse(
+        onlineFirst && !options.offline
+          ? dependencyProvider.solveOnline(options, elmVersion, elmJson, extra)
+          : dependencyProvider.solveOffline(options, elmVersion, elmJson, extra)
+      )
     );
+
+    return /** @type {ApplicationDependencies} */ (dependencies);
```

Comment:
Just so I understand, the `/** @type {unknown} */` is for correctly annotating the value of `dependencies`, and the type annotation here is to cast?

Or are they both for casting? In which case, why not cast only once?

Reply:
When using `JSON.parse` (& other lib functions), you generally want to cast to unknown, then to another type (the `as unknown as` pattern), because they don't default to `unknown` for back compat.
I'd guess it's not a direct return b/c of `no-unsafe-returns`.